dotCMS/src/main/java/com/dotcms/csspreproc/dartsass/config/SassCommandStrategyResolver.java

```diff
@@ -0,0 +1,157 @@
+    private List<SassCommandStrategy> getDefaultStrategies() {
+        final ImmutableList.Builder<SassCommandStrategy> builder = new ImmutableList.Builder<>();
+        builder.add(new MacOSSassCommandStrategyImpl());
+        builder.add(new LinuxSassCommandStrategyImpl());
+        return builder.build();
+    }
+
+    /**
+     * Adds new SASS Command Strategy to the current list, in case a new Operating System and or architecture must be
+     * supported.
+     *
+     * @param strategy The new {@link SassCommandStrategy}.
+     */
+    public synchronized void subscribe(final SassCommandStrategy strategy) {
+        if (null != strategy) {
+            final ImmutableList.Builder<SassCommandStrategy> builder = new ImmutableList.Builder<>();
+            builder.addAll(this.strategies);
+            builder.add(strategy);
+            this.strategies = builder.build();
+        }
+    }
+
+    /**
+     * Returns an Optional with the appropriate SASS Command Strategy based on the current environment. Considering this
+     * is an automatic process, there's no need for parameters or any other user input for dotCMS to correctly determine
+     * what specific command must be used for your current Operating System and architecture.
+     *
+     * @return The appropriate {@link SassCommandStrategy} wrapped into an Optional.
+     */
+    public Optional<SassCommandStrategy> getStrategy() {
+        if (this.selected.isPresent()) {
+            return this.selected;
+        }
+        for (final SassCommandStrategy strategy : this.strategies) {
+            if (strategy.test()) {
+                this.selected = Optional.of(strategy);
+                return this.selected;
+            }
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Assembles the appropriate Dart SASS compiler command for Mac OS X environments and their respective architectures.
+     */
+    private class MacOSSassCommandStrategyImpl implements SassCommandStrategy {
+
+        private final Map<String, String> DART_SASS_CMD = Map.of(
+                RuntimeUtils.ARM64_ARCH, "dart-sass-macos-arm64/sass",
+                RuntimeUtils.X86_64_ARCH, "dart-sass-macos-x64/sass");
+
+        @Override
+        public boolean test() {
+            if (SystemUtils.IS_OS_MAC) {
+                if (SystemUtils.OS_ARCH.equalsIgnoreCase(RuntimeUtils.ARM64_ARCH) || SystemUtils.OS_ARCH.equalsIgnoreCase(RuntimeUtils.X86_64_ARCH)) {
+                    return Boolean.TRUE;
+                } else {
+                    Logger.error(this, String.format("Dart SASS lib for Mac OS is not available for the specified " +
+                                                             "arch '%s'", SystemUtils.OS_ARCH));
+                }
+            }
+            return Boolean.FALSE;
+        }
+
+        @Override
+        public String apply() {
+            final String location = DART_SASS_LOCATION.get().endsWith(File.separator) ? DART_SASS_LOCATION.get() :
```

Comment:
Shouldn't this `endsWith` evaluation be part of `DART_SASS_LOCATION` definition?

Reply:
Resolved with Victor